### PR TITLE
Mat 7611 result date time not shifted

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR Template

Jira Ticket: [MAT-7611](https://jira.cms.gov/browse/MAT-7611)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [ ] Required tests are included.
* [x] No extemporaneous files are included (i.e. Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/models/cqm/datacriteria/basetypes/DataElement.java [hunk 1]
```diff
@@ -138,4 +144,49 @@ public ZonedDateTime shiftDateByYear(ZonedDateTime dateTime, int year) {
...
+    if (object == null) {
+      return null;
+    }
+
+    Object convertedObj = object;
+    // Date type
+    try {
+      LocalDate converted = convertToLocalDate(object);
+      if (converted != null) {
+        converted = converted.plusYears(year);
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/models/cqm/datacriteria/basetypes/DataElement.java:158]: Need to limit the resulting year range to [1900, 9999] to avoid display issues.

**Review by jkotanchik-SB — CHANGES_REQUESTED** [review]
Limit year result range to [1900, 9999].

File: src/main/java/gov/cms/madie/models/cqm/datacriteria/basetypes/DataElement.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/cqm/datacriteria/basetypes/DataElement.java:158]: @jkotanchik-SB Thanks so much Joe! I added the limitation.

**Review by jkotanchik-SB — APPROVED** [review]
